src/GraphQL.NewtonsoftJson/GraphQLRequestListJsonConverter.cs

```diff
@@ -60,9 +66,8 @@ public override object ReadJson(JsonReader reader, Type objectType, object? exis
                         : list;
                 }
 
-                var request = serializer.Deserialize<GraphQLRequest>(reader);
-                if (request != null)
-                    list.Add(request);
+                var request = serializer.Deserialize<GraphQLRequest>(reader)!;
```

Comment:
Does `request` always `null`?

Reply:
No -- it will be null in the example shown at the top of this PR.

Reply:
So list of non-nullable requests will contain `null` here but later this `null` will be properly handled, right? I doubt since I see these lines in server:
```
await RequestExecutingAsync(gqlRequestInBatch, i);
var result = await ExecuteRequestAsync(gqlRequestInBatch, userContext, executer, context.RequestServices, cancellationToken);
```

`gqlRequestInBatch` should be `null` if I understand correctly. So eventually code will throw NRE here:
```
 protected virtual ExecutionOptions GetOptions(GraphQLRequest request, IDictionary<string, object> context, IServiceProvider requestServices, CancellationToken cancellationToken)
        {
            var opts = new ExecutionOptions
            {
                Schema = Schema,
                OperationName = request.OperationName, <--------------------- BOOM
                Query = request.Query,
                Variables = request.Variables,
                Extensions = request.Extensions,
                UserContext = context,
                CancellationToken = cancellationToken,
                ComplexityConfiguration = _options.ComplexityConfiguration,
                EnableMetrics = _options.EnableMetrics,
                UnhandledExceptionDelegate = _options.UnhandledExceptionDelegate,
                MaxParallelExecutionCount = _options.MaxParallelExecutionCount,
                RequestServices = requestServices,
            };

            return opts;
        }
```

Reply:
I agree.  I've already fixed this in my copy of the server project and added a test for it.